Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -1,500 +1,487 @@
             {
-                _logger.NotifyError(BatchContext, string.Format("Could not clean up temporary destination file '{0}': {1}", DestinationFileDuringTransfer, ex.Message), ex);
+                Trace(TransferState.RemoveTemporaryDestinationFile, "Removing temporary destination file {0}", DestinationFileDuringTransfer);
+                File.Delete(DestinationFileDuringTransfer);
             }
         }
+        catch (Exception ex)
+        {
+            _logger.NotifyError(BatchContext, string.Format("Could not clean up temporary destination file '{0}': {1}", DestinationFileDuringTransfer, ex.Message), ex);
+        }
+    }
+
+    private void TryToRemoveSourceTempFile(string filePath)
+    {
+        try
+        {
+            if (FileDefinedAndExists(filePath)) Client.Delete(filePath);
+        }
+        catch (Exception e)
+        {
+            _logger.NotifyError(BatchContext, string.Format("Could not clean up local file '{0}'", Path.GetFileName(filePath)), e);
+        }
+    }
+
+    private bool FileDefinedAndExists(string filePath)
+    {
+        return !string.IsNullOrEmpty(filePath) && Client.Exists(filePath);
+    }
 
-        private void TryToRemoveSourceTempFile(string fileName)
+    private string RestoreSourceFileAfterErrorIfItWasRenamed()
+    {
+        // restore the source file so we can retry the operations
+        // - but only if the source file has been renamed in the first place
+        if (!string.IsNullOrEmpty(SourceFileDuringTransfer))
         {
             try
             {
-                if (FileDefinedAndExists(fileName))
+                if (ShouldSourceFileBeRestoredOnError())
                 {
-                    Client.GetAttributes(fileName).SetPermissions(777);
-                    Client.Delete(fileName);
+                    Client.RenameFile(SourceFileDuringTransfer, SourceFile.FullPath);
+                    return "[Source file restored.]";
                 }
             }
-            catch (Exception e)
+            catch (Exception ex)
             {
-                _logger.NotifyError(BatchContext, string.Format("Could not clean up local file '{0}'", fileName), e);
+                var message = string.Format(
+                    "Could not restore original source file '{0}' from temporary file '{1}'. Error: {2}.",
+                    Path.GetFileName(SourceFile.FullPath),
+                    Path.GetFileName(SourceFileDuringTransfer),
+                    ex.Message);
+
+                _logger.NotifyError(BatchContext, message, ex);
+                return string.Format("[{0}]", message);
             }
         }
 
-        private bool FileDefinedAndExists(string filename)
-        {
-            return !string.IsNullOrEmpty(filename) && Client.Exists(filename);
-        }
+        return string.Empty;
+    }
 
-        private string RestoreSourceFileAfterErrorIfItWasRenamed()
+    private string RestoreDestinationFileAfterErrorIfItWasRenamed(SftpClient client)
```

Comment:
You are not using parameter `client`, so it can be removed.

Reply:
Removed.